### PR TITLE
migrate images to docker hub 

### PR DESCRIPTION
**Pull request by prydonius**
xref #1912 

Signed-off-by: Adnan Abdulhussein <aadnan@vmware.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
-      all_branches: true
+- provider: script
+  skip_cleanup: true
+  script: hack/docker-push.sh
+  on:
+    repo: vmware-tanzu/velero
+    all_branches: true
+env:
+  global:
+  - secure: Sa0R7bhOvrybIWvURjEpOLQ3/v1IqiSM2uwYTSMhLtwHunkiYXpbXi/wVPkujM7kgnFeJzGfNTZI6blkl3Vnz6Ca48avxiv+KRk7bNlTn/e2LkJaqOh/NcwqrVyWG8cZpWTHAzwJkHrV6xcWRTWFqx/UhUHH4PdmXd0pj3/DDxHhUZXJkZfYDC2uI+kmhJd9x1yxxz6OTXrGeMw22R5RtgltcQYjYWjGhXjNJ96+0QhC6juCwwtuU3oz7vfV1dP6ea9yeiHKCdDR0xp/Ymtlt4DULgQv4vuVBCR7LuPjsdLI1QslKbCz+vk/XwEcckf3R+yJUaArGLSM0f3AwuAZnETwkfKu9gJCEz8Xvvwsy5QHj/DGeSr+JY4Ylb1YzJG48R5QMcNwOgQOpewamBggxkyyOlNanUGzMJ8F7zjDOZjOUsRSSG+cn01/oJEHNyFmNkxbugOGuxqpi8qIfyWrdPSb8AdnUVUi2GDR/unSm2uZLSJtjaeypqzxalFdL04pbkk6/vKZTi6VMIyewHeR0XIC/VKQ+av2CwjnfqTtTHQHMh2kVxVzy4SKPYgWvbUUg9SG0z2lmfBWzE1HS+ozUKruxBu/YnPHaFz3D/+w0AwtHSYp3FMkCC/2SZU1hiOzAKX+S8T5K1OKoBiIEEUVa90EnSslCIiNIUHXgdtG62U=
```
Comment (skriss) [inline .travis.yml]: how are these generated? let's add a comment

**Comment by skriss** [comment]
We also have `gcr.io/heptio-images` hardcoded in a few different places (example YAML, `pkg/install`, pkg/restore` are the ones I see on a quick scan). We'll need to update those prior to the release. Since many of them refer to the `latest` tag we'll want to get v1.0, v1.1/latest pushed up to docker hub too.

Can tackle that in a separate PR once we confirm this is working.

**Comment by prydonius** [comment]
Thanks @skriss, I think a separate PR makes the most sense. I've captured the other TODO items in https://github.com/vmware-tanzu/velero/issues/1912#issuecomment-542416056

File: .travis.yml [hunk 2]
```diff
@@ -1,31 +1,23 @@
...
 sudo: required
-
 services:
-  - docker
-
+- docker
 script:
-  - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
-  - echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, PR=$TRAVIS_PULL_REQUEST, BRANCH=$BRANCH"
-  - hack/ci-check.sh
```
Comment (skriss) [inline .travis.yml]: Hmm, looks like the `hack/ci-check.sh` script step was dropped, so we're not getting a `make ci` run before moving on to building/pushing container images -- was that intentional? I'm thinking we probably want to retain it, so we get all units run.

**Pull request merged**